src/main/java/com/google/cloud/tools/gradle/appengine/core/DeployExtension.java

```diff
@@ -58,6 +65,65 @@ public DeployExtension(DeployExtension deployExtension) {
     this.stopPreviousVersion = deployExtension.stopPreviousVersion;
     this.version = deployExtension.version;
     this.appEngineDirectory = deployExtension.appEngineDirectory;
+    this.appEngineWebXml = deployExtension.appEngineWebXml;
+  }
+
+  /**
+   * Verifies that project/version properties are pulled correctly from build.gradle and
+   * appengine-web.xml and modifies the properties appropriately.
+   *
+   * @return A copy of the DeployExtension with the corrected properties.
+   */
+  public DeployExtension withPropertiesFromAppEngineWebXml()
+      throws AppEngineException, IOException, SAXException {
+    DeployExtension deployExtensionCopy = new DeployExtension(this);
+
+    if (appEngineWebXml == null) {
+      return deployExtensionCopy;
+    }
+
+    AppEngineDescriptor appengineWebXmlDoc =
+        AppEngineDescriptor.parse(new FileInputStream(appEngineWebXml));
+    String xmlProject = appengineWebXmlDoc.getProjectId();
+    String xmlVersion = appengineWebXmlDoc.getProjectVersion();
+
+    // Verify that project and version are set somewhere
+    if (project == null && xmlProject == null || version == null && xmlVersion == null) {
```

Comment:
Now that I think about it, I think version can be autogenerated by the cloud sdk.

Reply:
we should find out how the IDEs handle this

Reply:
Yeah, it looks like it can. So I guess we should do the same validation to avoid conflicting build config/appengine-web.xml, but skip the error if it's not configured.